compiler/rustc_resolve/src/ident.rs

```diff
@@ -870,6 +870,7 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                         dedup_span: path_span,
                         outermost_res: None,
                         parent_scope: *parent_scope,
+                        imported,
```

Comment:
```suggestion
                        single_nested: path_span != root_span,
```
Looks like `Finalize` doesn't need to carry additional data (especially with a lifetime), the property we need can be inferred from already existing fields.
Also `PrivacyError` doesn't need to keep whole `Option<Import<'a>>`, it only needs a single boolean flag.